build/utils.cake

```diff
@@ -0,0 +1,201 @@
+    FixForMono(settings, "ILRepack.exe");
+    ILRepack(outFilePath, targetPath, sourceFiles, settings);
+
+    CopyFileToDirectory("./LICENSE", ilMergeDir);
+    CopyFileToDirectory(targetDir + "GitVersion.pdb", ilMergeDir);
+
+    Information("Copying libgit2sharp files..");
+
+    if (!includeLibGit2Sharp) {
+        CopyFileToDirectory(targetDir + "LibGit2Sharp.dll", ilMergeDir);
+    }
+    CopyFileToDirectory(targetDir + "LibGit2Sharp.dll.config", ilMergeDir);
+    CopyDirectory(targetDir + "/lib/", ilMergeDir + "/lib/");
+}
+
+void PublishILRepackedGitVersionExe(bool includeLibGit2Sharp, DirectoryPath targetDir, DirectoryPath ilMergDir, DirectoryPath outputDir, string configuration, string dotnetVersion)
+{
+    ILRepackGitVersionExe(includeLibGit2Sharp, targetDir, ilMergDir);
+    CopyDirectory(ilMergDir, outputDir);
+
+    if (includeLibGit2Sharp) {
+        CopyFiles("./src/GitVersionExe/NugetAssets/*.ps1", outputDir);
+    }
+
+    // Copy license & Copy GitVersion.XML (since publish does not do this anymore)
+    CopyFileToDirectory("./LICENSE", outputDir);
+    CopyFileToDirectory("./src/GitVersionExe/bin/" + configuration + "/" + dotnetVersion + "/GitVersion.xml", outputDir);
+}
+
+void DockerBuild(GitVersion version, string platform, string variant)
+{
+    var name = $"gittools/gitversion-{variant}";
+    var workDir = DirectoryPath.FromString($"./src/Docker/{platform}/{variant}");
+
+    DirectoryPath sourceDir;
+    if (variant == "dotnetcore") {
+        sourceDir = parameters.Paths.Directories.ArtifactsBinNetCore.Combine("tools");
+    } else {
+        sourceDir = parameters.Paths.Directories.ArtifactsBinFullFxCmdline.Combine("tools");
+    }
+    CopyDirectory(sourceDir, workDir.Combine("content"));
+
+    var buildSettings = new DockerImageBuildSettings
+    {
+        Rm = true,
+        Tag = new []{ $"{name}:{platform}-{version.MajorMinorPatch}" },
+        File = $"{workDir}/Dockerfile",
+        BuildArg = new []{ $"contentFolder=/content" },
+        Pull = true,
+        // Platform = platform // TODO this one is not supported on docker versions < 18.02
+    };
+
+    DockerBuild(buildSettings, workDir.ToString());
+}
+
+void DockerPush(GitVersion version, string platform, string variant)
+{
+    var name = $"gittools/gitversion-{variant}";
+    var tag = $"{name}:{platform}-{version.MajorMinorPatch}";
+
+    Information("Tag Name {0}", tag);
+    DockerPush(tag);
+}
+
+void GetReleaseNotes(FilePath outputPath, DirectoryPath workDir, string repoToken)
+{
+    var toolPath = Context.Tools.Resolve("GitReleaseNotes.exe");
+
+    var arguments = new ProcessArgumentBuilder()
+                .Append(workDir.ToString())
+                .Append("/OutputFile")
+                .Append(outputPath.ToString())
+                .Append("/RepoToken")
+                .Append(repoToken);
+
+    StartProcess(toolPath, new ProcessSettings { Arguments = arguments, RedirectStandardOutput = true }, out var redirectedOutput);
```

Comment:
First I have tried to use the cake alias for that, for some reason locally it was working, on build servers - AppVeyor and Azure windows agents, it was crashing, but when calling with StartProcess was working. I guess after we finish with this PR I'll get back to this issue and try to use the alias

Reply:
Hmm, that is very interesting. Would be great to get to the bottom of that, as it might be that we need to change something in Cake itself (which is where the alias lives).  As you say, this shouldn't stop the PR from moving forward, but something to circle back around to looking at.

Reply:
https://ci.appveyor.com/project/arturcic/gitversion-ee285/build/4.0.0-build-scripts.1+1626.build.267#L205 this is the error when using the alias

Reply:
@arturcic would be good to run that with Diagnostic Verbosity to "see" what command that Cake is sending to the command line.  Could be that there is a bug in there.

Reply:
thanks for suggestion, I'll do that as part of a new PR for using the alias